WeakAuras/Types.lua

```diff
@@ -1332,20 +1332,47 @@ Private.item_quality_types = {
   [8] = ITEM_QUALITY8_DESC,
 }
 
-Private.discovered_currencies = {}
-Private.discovered_currencies_sorted = {}
-for index = 1, C_CurrencyInfo.GetCurrencyListSize() do
-  local CurrencyInfo = C_CurrencyInfo.GetCurrencyListInfo(index)
-  local currencyLink = C_CurrencyInfo.GetCurrencyListLink(index)
-  if currencyLink then
-    local currencyID = C_CurrencyInfo.GetCurrencyIDFromLink(currencyLink)
-    local icon = CurrencyInfo.iconFileID or "Interface\\Icons\\INV_Misc_QuestionMark" --iconFileID not available on first login
-    Private.discovered_currencies[currencyID] = ("|T%s:16|t %s"):format(icon, CurrencyInfo.name)
-    Private.discovered_currencies_sorted[currencyID] = index
+local function InitializeCurrencies()
```

Comment:
At first login, the currencies seem to be unavailable for me. I made the currency tables all lazy initialized whenever the options open them.

There's a small bug where we don't update the list if the user learns a new currency, though I'm not sure what event fires for that.

Reply:
There is `CURRENCY_DISPLAY_UPDATE` but that also fires when any currency changes, including currencies that are always hidden, like [renown](https://wago.tools/db2/CurrencyTypes?page=1&filter[ID]=2653). 
If a user discovers a new currency during their session they can fallback to the specific currency option or reload.